torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -257,3 +263,46 @@ def lexicon_decoder(
         sil_token=sil_token,
         unk_word=unk_word,
     )
+
+
+def download_pretrained_files(model: str) -> PretrainedFiles:
+    """
+    Retrieves pretrained data files used for CTC decoder.
+
+    Args:
+        model (str): pretrained language model to download
+            Options: ["librispeech-3-gram", "librispeech-4-gram", "librispeech"]
+
+    Returns:
+        Dictionary with the following key mapping:
+            lm: path corresponding to downloaded language model, or None if lm not provided
+            lexicon: path corresponding to downloaded lexicon file
+            tokens: path corresponding to downloaded tokens file
+    """
+
+    hub_dir = torch.hub.get_dir()
+    path = f"{hub_dir}/{model}"
+
+    url_prefix = "https://download.pytorch.org/torchaudio/decoder-assets"
+    model_dir = f"{url_prefix}/{model}"
+
+    tokens_file = f"{path}/tokens.txt"
+    lexicon_file = f"{path}/lexicon.txt"
+
+    download_url_to_file(f"{model_dir}/tokens.txt", tokens_file)
+    download_url_to_file(f"{model_dir}/lexicon.txt", lexicon_file)
+
+    # Support for decoding without a language model
+    lm_url = f"{model_dir}/lm.bin"
+    response = requests.get(lm_url)
+    if response.status_code == 200:
+        lm_file = f"{path}/lm.bin"
+        download_url_to_file(lm_url, lm_file)
+    else:
+        lm_file = None
```

Comment:
This does not handle the case where the model exist but the download failed for some network issue.
`lm_file=None` should not be the result of download but the information encoded in this code.

See the comment above.